falcon/api.py

```diff
@@ -532,6 +533,31 @@ def add_error_handler(self, exception, handler=None):
         the client.  Alternatively, a handler may modify `resp`
         directly.
 
+        An error handler "matches" a raised exception if the exception is an
+        instance of the corresponding exception type. If more than one error
+        handler matches the raised exception, the framework will choose the
+        most specific one. If multiple error handlers are registered for the
+        *same* exception class, then the most recently-registered handler is
+        used. For example, suppose we register error handlers as follows::
+
+            app = falcon.API()
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_not_found)
+            app.add_error_handler(falcon.HTTPError, custom_handle_http_error)
+            app.add_error_handler(Exception, custom_handle_uncaught_exception)
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_404)
+
+        QUESTION(csojinb): Do we want the usages of falcon.HTTP* below to include links to the error classes?
+
+        If an instance of ``falcon.HTTPForbidden`` is raised, it will be
```

Comment:
Open questions on the style in this section:

* Should I use single or double backticks for the handler function names? I couldn't decide whether they counted as "variables" or "inline code".
* Should the ussages of `falcon.HTTP*` in this example include `:class:` so that there's a link to the relevant part of the api doc? I didn't do that to start because the specific error classes used for the examples are fairly arbitrary.